better_exceptions/formatter.py

```diff
@@ -110,7 +111,61 @@ def append(offset, node, s, theme):
         return self.colorize_comment(''.join(chunks))
 
     def get_relevant_names(self, source, tree):
-        return [node for node in ast.walk(tree) if isinstance(node, ast.Name)]
+        return [node for node in ast.walk(tree) if isinstance(node, (ast.Name, ast.Attribute))]
+
+    def get_nested_attribute(self, name_or_attribute, frame):
+        node = name_or_attribute
+        col = node.col_offset
+        first_text = None
+        text = None
+        last_col = None
+        val = None
+        attrs = [] # list of dict containing intermediate accessors
+        cpt = 0
+        while node:
+            if isinstance(node, ast.Attribute):
+                # get attr, return value
+                text = node.attr
+                last_col = len('.'+text)
+                col += last_col
+                attrs.append(dict(type='attribute', attr=text, col=last_col))
+                node = node.value
+            elif isinstance(node, ast.Call):
+                # @todo: get fun/args, return fun
+                pass
+            elif isinstance(node, ast.Subscript):
+                # @todo: get slice.value.(num|id), return value
+                pass
+            elif isinstance(node, ast.Name):
+                text = node.id
+                last_col = len(text)
+                col += last_col
+                if text in frame.f_locals:
+                    val = frame.f_locals.get(text, None)
+                elif text in frame.f_globals:
+                    val = frame.f_globals.get(text, None)
+
+                node = None
+            else:
+                node = None
+
+            if cpt == 0 and text:
+                first_text = text
+
+            cpt += 1
+
+        if not val or len(attrs) == 0:
+            col -= last_col
+            return text, col, val
+
+        for n_attr, attr in enumerate(attrs[::-1]):
+            if attr['type'] == 'attribute':
+                if n_attr == len(attrs)-1:
+                    col -= attr['col']-1
+                #val = inspect.getattr_static(val, attr['attr'], '__empty') # py3
```

Comment:
`getattr_static` or some equivalent definitely needs to be used, not just raw `getattr`. What's going on here?

Reply:
I commented the line with  `getattr_static` and used the raw `getattr` instead only because it is not compatible/existing on python2. But Iam not conservative on this choice...

Reply:
It was decided in #67 to drop support for Python 2 at some point, although I don't know how that affects PRs at the moment.

Reply:
Positively. Please assume Python 3.

Reply:
Ok, I committed. (I leave the `PY3` condition to pass the test_all.sh, still the python-nightly didnt pass, I don't get why?)